protocol/server.go

```diff
@@ -1,57 +1,68 @@
 package protocol
 
 import (
+	"context"
+	"io"
+
 	"github.com/domainr/epp2/protocol/dataunit"
 	"github.com/domainr/epp2/schema"
 	"github.com/domainr/epp2/schema/epp"
 )
 
-// Writer is the interface implemented by any type that can write an EPP message body.
-type Writer interface {
-	WriteEPP(epp.Body) error
+// Responder is the interface implemented by any type that can respond to a client request with an EPP message body.
+type Responder interface {
+	// RespondEPP writes an EPP response to the client. It blocks until the message is written,
+	// Context is cancelled, or the underlying connection is closed.
+	//
+	// RespondEPP can be called from an arbriary goroutine, but must only be called once.
+	RespondEPP(context.Context, epp.Body) error
 }
 
-type writerFunc func(body epp.Body) error
+type responderFunc func(context.Context, epp.Body) error
 
-func (f writerFunc) WriteEPP(body epp.Body) error {
-	return f(body)
+func (f responderFunc) RespondEPP(ctx context.Context, body epp.Body) error {
+	return f(ctx, body)
 }
 
-// Server is a low-level server for the Extensible Provisioning Protocol (EPP)
-// as defined in [RFC 5730]. A Server is safe to use from multiple goroutines.
+// Server represents a a low-level EPP server connection, as defined in [RFC 5730].
+// A Server is safe to use from multiple goroutines.
 //
 // [RFC 5730]: https://datatracker.ietf.org/doc/rfc5730/
 type Server interface {
 	// ServeEPP provides an client EPP request and a mechanism to respond to the request.
-	// It blocks until a response is received or the underlying connection is closed.
-	// The returned [Writer] should only be used once. The returned Writer will always
+	// It blocks until a response is received, Context is canceled, or the underlying connection is closed.
```

Comment:
Is this blocking not until a response is sent/returned to the client? I didn't think the server would wait for the client to read (beyond a timeout) from the server and respond to it.